mibidata/tests/test_tiff.py

```diff
@@ -262,15 +265,93 @@ def test_write_single_channel_tiffs(self):
 
             np.testing.assert_equal(np.squeeze(tif.data), DATA[:, :, i])
             self.assertTupleEqual(tif.channels, (CHANNELS[i], ))
-            self.assertEqual(tif.data.dtype, np.uint16)
+            self.assertEqual(tif.data.dtype, np.float32)
+
+    def test_tiff_dtype_correct_arguments(self):
+        supported_dtypes = [np.float32, np.uint16]
+        for dtype in supported_dtypes:
+            tiff.write(self.filename, self.float_image, multichannel=True,
+                       dtype=dtype)
+
+        unsupported_types = ['abcdef', np.str, np.bool, np.uint8, np.float64]
+        for dtype in unsupported_types:
+            with self.assertRaises(ValueError):
+                tiff.write(self.filename, self.float_image, multichannel=True,
+                           dtype=dtype)
+
+    def test_write_float_is_deprecated(self):
+        with self.assertRaises(ValueError):
+            tiff.write(self.filename, self.int_image, write_float=True)
+            tiff.write(self.filename, self.float_image, write_float=False)
```

Comment:
This only tests that one of these statements raises. And I think testing once is fine, but am noting that having the second statement doesn't do anything.

Reply:
Good catch. The second one actually fails when tested individually because I only tested `if write_float` and not that it isn't None. Since I know I have places in code where I pass `write_float=False`, I am making sure that each case is tested.